File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/core/converter/DateConverterTest.java
package coolpharaoh.tee.speicher.tea.timer.core.converter;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

import java.util.Date;

import coolpharaoh.tee.speicher.tea.timer.core.date.DateConverter;

class DateConverterTest {
    public static final long LONG_DATE = 214234L;

    @Test
    void convertLongToDate() {
        final Date dateAfter = DateConverter.toDate(LONG_DATE);

        assertThat(dateAfter).isEqualTo(new Date(LONG_DATE));
    }

    @Test
    void convertNullToDateReturnNull() {
        final Date dateAfter = DateConverter.toDate(null);

        assertThat(dateAfter).isNull();
    }

    @Test
    void convertDateToLong() {
        final Date date = new Date(LONG_DATE);

        final Long longAfter = DateConverter.fromDate(date);

        assertThat(longAfter).isEqualTo(LONG_DATE);
    }

    @Test
    void convertNullToLongReturnNull() {
        final Long longAfter = DateConverter.fromDate(null);

        assertThat(longAfter).isNull();
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/core/converter/DateConverterTest.kt
package coolpharaoh.tee.speicher.tea.timer.core.converter

import coolpharaoh.tee.speicher.tea.timer.core.date.DateConverter.fromDate
import coolpharaoh.tee.speicher.tea.timer.core.date.DateConverter.toDate
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.util.Date

internal class DateConverterTest {
    @Test
    fun convertLongToDate() {
        val dateAfter = toDate(LONG_DATE)

        assertThat(dateAfter).isEqualTo(Date(LONG_DATE))
    }

    @Test
    fun convertNullToDateReturnNull() {
        val dateAfter = toDate(null)

        assertThat(dateAfter).isNull()
    }

    @Test
    fun convertDateToLong() {
        val date = Date(LONG_DATE)

        val longAfter = fromDate(date)

        assertThat(longAfter).isEqualTo(LONG_DATE)
    }

    @Test
    fun convertNullToLongReturnNull() {
        val longAfter = fromDate(null)

        assertThat(longAfter).isNull()
    }

    companion object {
        const val LONG_DATE = 214234L
    }
}